script/src/test/java/io/kestra/plugin/scripts/runner/docker/DockerTest.java

```diff
@@ -37,4 +37,29 @@ void shouldNotHaveTagInDockerPullButJustInWithTag() throws Exception {
         assertThat(result.getExitCode()).isZero();
         Assertions.assertThat(result.getLogConsumer().getStdOutCount()).isEqualTo(1);
     }
+
+    @Test
+    void shouldSetCorrectCPULimitsInContainer() throws Exception {
+        var runContext = runContext(this.runContextFactory);
+
+        var cpuConfig = Cpu.builder()
+            .cpus(Property.ofValue(1.5))
+            .build();
+
+        var docker = Docker.builder()
+            .image("rockylinux:9.3-minimal")
+            .cpu(cpuConfig)
+            .build();
+
+        var taskCommands = new CommandsWrapper(runContext).withCommands(Property.ofValue(List.of(
+            "/bin/sh", "-c",
+            "cat /sys/fs/cgroup/cpu.max || cat /sys/fs/cgroup/cpu/cpu.cfs_quota_us"
+        )));
+
+        var result = docker.run(runContext, taskCommands, Collections.emptyList());
+
+        assertThat(result).isNotNull();
```

Comment:
I assume you verify with your eyes that the result is correct ;) 
Another solution would have been to create outputs so you would be able to assert on it.
Or even check the logs as you have them inside the log consumer.

I'll approve it anyway but more robust test assertions would be great.